blog-post/blog-post.md

```diff
@@ -0,0 +1,738 @@
+
+```solidity
+(bytes memory a, bytes memory b) = abi.decode(input, (bytes, bytes))
+```
+
+The same definition is much cleaner in Core:
+
+```solidity
+let (a, b) = abi.decode(input, (uint256, uint256))
+```
+
+Another common frustration with Classic Solidity is the syntactic noise required when defining array
+literals. Consider the following snippet:
+
+```Solidity
+uint256[3] memory a = [1, 2, 3];
+```
+
+This declaration is rejected by the compiler with the following error message:
+
+```
+Error: Type uint8[3] memory is not implicitly convertible to expected type uint256[3] memory.
+```
+
+Classic Solidity implements a limited and special cased form of type inference for numeric array
+literals: it selects the smallest possible element type that can represent every element of the
+array (in this case `uint8`). The compiler then throws a type error when attempting to assign this
+value to a variable with a different type.
+
+In order for the previous definition be accepted, we can add a type coercion to the first array element:
+
+```solidity
+uint256[3] memory a = [uint256(1), 2, 3];
+```
+
+The constraint based inference algorithm in Core Solidity is a lot more general, and will allow us
+to omit this coercion:
+
+```solidity
+uint256[3] memory a = [1, 2, 3];
+```
+
+### Compile Time Evaluation
+
+We do not yet have a prototype implementation of compile time evaluation, so don't have concrete
+examples to share here yet. We are however very convinced that this will be a particularly valuable
+extension to the language and consider it a critical feature that needs to be in place before
+release. A strong goal is to minimize the differences between the runtime and compile time variants
+of the language, allowing for a familiar syntax and code sharing between the two contexts.
+
+Non-trivial design and implementation work remains here. We are exploring the degree to which access
+to memory at compile time is required, and if it is, what kind of analysis passes we would want to
+implement to guard against accidental leakage of references to compile time memory. We are also
+investigating what kind of compile time specific primitives we might want to add, and whether we
+want to expand the languages capabilities around reflection.
+
+We will publish more on our designs once they stabilise. We want to make sure the needs of the
+community are met here: if you have concrete real world use cases in mind for this feature, we would
+be very interested to hear them.
+
+## SAIL, Desugaring and the Standard Library
+
+In addition to expanding the surface language, the transition to Core Solidity will also introduce a
+new user accessible mid level IR: SAIL (Solidity Algebraic Intermediate Language). This is the
+"Core" in Core Solidity. It's the most minimal language we could conceive of that would let us
+express the full range of high-level language constructs found in Classic Solidity. It can usefully
```

Comment:
```suggestion
"Core" in Core Solidity. It's the most minimal language that lets us
express the full range of high-level language constructs found in Classic Solidity. It can usefully
```